### PR TITLE
Throw exception when an empty string is passed to the constructor

### LINKED ISSUE #124: Empty Color constructor returns black value
When calling Color with `Color('')` it returns a black value. It feels like this should probably throw an error of an unrecognised string. Is this a known issue?

### PR DESCRIPTION
**Pull request by narrator0**
resolve #124

### CODE REVIEW

**Comment by narrator0** [comment]
@Qix- you suggest to throw when color-string returns null but it actually never goes to color-string because `!obj` is true when `obj === ''`. So added an extra condition.

File: index.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ function Color(obj, model) {
 	var i;
 	var channels;
 
-	if (!obj) {
+	if (typeof obj === 'undefined') {
```
Comment (narrator0) [inline index.js]: @Qix-  I changed this condition to allow empty string to be deal by color-string. However, I am not sure if there are conditions other than undefined that need to be dealt here.

File: index.js [hunk 2]
```diff
@@ -50,7 +50,7 @@ function Color(obj, model) {
 	} else if (typeof obj === 'string') {
 		var result = colorString.get(obj);
 		if (result === null) {
-			throw new Error('Unable to parse color from string: ' + obj);
+			throw new Error('Unable to parse color from string: "' + obj + '"');
```
Comment (Qix-) [inline index.js]: Without the double quotes, please

File: index.js (hunk 1 above)
Comment (Qix-) [reply index.js]: Aha yes. I see now, this is correct.

File: index.js (hunk 2 above)
Comment (narrator0) [reply index.js]: Got it. Just dropped that commit.

**Comment by Qix-** [comment]
Great :) Thank you so much!

**Comment by Qix-** [comment]
Released as `3.1.0`.

**Pull request merged**